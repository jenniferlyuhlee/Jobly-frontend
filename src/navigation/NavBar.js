import React, { useContext } from "react";
import { Link, NavLink } from "react-router-dom";
import UserContext from "../auth-user/UserContext";
import "./NavBar.css"

function NavBar () {
    const {currUser, logout} = useContext(UserContext);

    return (
        <div className="NavBar">
            <div className="NavBar-left">
                <NavLink className="NavBar-link-left" 
                    to="/">Jobly
                </NavLink>
            </div>
            <div className="NavBar-right">
                {currUser?
                // if user, show logged in navbar
                <div>
                    <NavLink className="NavBar-link-right" 
                    to="/companies">Companies
                    </NavLink>
                    <NavLink className="NavBar-link-right" 
                        to="/jobs">Jobs
                    </NavLink>
                    <NavLink className="NavBar-link-right" 
                        to="/profile">Profile
                    </NavLink>
                    <Link className="NavBar-link-right" 
                        to="/" 
                        onClick={logout}>Logout
                    </Link>  
                </div>  
                :
                // if not user, show default navbar
                <div>
                    <Link className="NavBar-link-right" 
                        to="/login">Login
                    </Link>
                    <Link className="NavBar-link-right" 
                        to="/signup">Sign Up
                    </Link>
                </div>
            }
            </div>
        </div>
    )
}

export default NavBar;