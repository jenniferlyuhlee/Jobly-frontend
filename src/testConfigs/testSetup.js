import React from "react";
import UserContext from "../auth-user/UserContext";

const demoUser = {
    username: "testuser",
    first_name: "testfirst",
    last_name: "testlast",
    email: "test@test.com",
    photo_url: null,
    applications: []
  };

  const UserProvider =
      ({ children, currUser = demoUser, hasAppliedToJob = () => false }) => (
      <UserContext.Provider value={{ currUser, hasAppliedToJob }}>
        {children}
      </UserContext.Provider>
  );

  export {UserProvider};
