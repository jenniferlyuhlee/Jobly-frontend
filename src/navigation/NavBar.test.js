import React from 'react';
import {render} from "@testing-library/react"
import { MemoryRouter } from 'react-router-dom';
import { UserProvider } from "../testConfigs/testSetup";
import NavBar from './NavBar';

it('renders without crashing', function() {
  render(
    <MemoryRouter>
        <UserProvider>
            <NavBar />
        </UserProvider>
    </MemoryRouter>
  )
});

it ('matches snapshot when logged in', function () {
    const {asFragment} = render(
        <MemoryRouter>
            <UserProvider>
                <NavBar />
            </UserProvider>
        </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
})

it ('matches snapshot when logged out', function () {
    const {asFragment} = render(
        <MemoryRouter>
            <UserProvider currentUser={null}>
                <NavBar />
            </UserProvider>
        </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
});
