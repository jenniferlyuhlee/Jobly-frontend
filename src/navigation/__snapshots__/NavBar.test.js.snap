// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot when logged in 1`] = `
<DocumentFragment>
  <div
    class="NavBar"
  >
    <div
      class="NavBar-left"
    >
      <a
        aria-current="page"
        class="NavBar-link-left active"
        href="/"
      >
        Jobly
      </a>
    </div>
    <div
      class="NavBar-right"
    >
      <div>
        <a
          class="NavBar-link-right"
          href="/companies"
        >
          Companies
        </a>
        <a
          class="NavBar-link-right"
          href="/jobs"
        >
          Jobs
        </a>
        <a
          class="NavBar-link-right"
          href="/profile"
        >
          Profile
        </a>
        <a
          class="NavBar-link-right"
          href="/"
        >
          Logout
        </a>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`matches snapshot when logged out 1`] = `
<DocumentFragment>
  <div
    class="NavBar"
  >
    <div
      class="NavBar-left"
    >
      <a
        aria-current="page"
        class="NavBar-link-left active"
        href="/"
      >
        Jobly
      </a>
    </div>
    <div
      class="NavBar-right"
    >
      <div>
        <a
          class="NavBar-link-right"
          href="/companies"
        >
          Companies
        </a>
        <a
          class="NavBar-link-right"
          href="/jobs"
        >
          Jobs
        </a>
        <a
          class="NavBar-link-right"
          href="/profile"
        >
          Profile
        </a>
        <a
          class="NavBar-link-right"
          href="/"
        >
          Logout
        </a>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
